acid-view/src/AcidView.hs

```diff
@@ -0,0 +1,45 @@
+module AcidView
+       (
+         loadState
+       , printDB
+       ) where
+
+import CmdParse
+import Universum
+
+import Ariadne.Wallet.Cardano.Kernel.DB.AcidState
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet
+
+import Data.Acid (AcidState(..), IsAcidic)
+import Data.Acid.Local (getState)
+import Data.Text.Buildable (build)
+import Data.Text.Lazy.Builder (Builder, toLazyText)
+
+import qualified Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet as IxSet
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Text.Lazy.Builder.Int as BI
+
+-- | Loads state of "pos" events ago from acid DB
+loadState :: (IsAcidic st) => FilePath -> st -> Int -> IO (Either String (AcidState st, BL.ByteString))
+loadState path initState pos = getState path initState pos True
+
+
+-- | Prints given part of DB to terminal of file.
+printDB :: TypeOfOutput -> Maybe FilePath -> DB -> IO ()
+printDB whatToPrint fileM db = do
+    let wallets     = db ^. dbHdWallets . hdWalletsRoots
+    let accounts    = db ^. dbHdWallets . hdWalletsAccounts
+    let addresses   = db ^. dbHdWallets . hdWalletsAddresses
+    let infoToPrint = case whatToPrint of
+                        Wallets   -> build wallets
+                        Accounts  -> build accounts
+                        Addresses -> build addresses
+                        AllData       -> build db
+    let additionalInfo = "\n\nTotal wallets: " <> (ixSize wallets) <> "\nTotal accounts: "
+                            <> (ixSize accounts) <> "\nTotal addresses: " <> (ixSize addresses)
+    case fileM of
+      Nothing   -> putStr . toLazyText $ infoToPrint <> additionalInfo
```

Comment:
Once again, try to use `putText`. I'm not sure any conversion is needed here.

Reply:
You steel need to make `Text` from `Builder`

Reply:
PutText needs strict Text, thus i shoul add one more conversion (toStrict). I think, that it would be better to use putStr.